service-base/src/main/java/org/eclipse/hono/service/monitoring/AbstractMessageSenderConnectionEventProducer.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+package org.eclipse.hono.service.monitoring;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+import org.eclipse.hono.client.HonoClient;
+import org.eclipse.hono.client.MessageSender;
+import org.eclipse.hono.service.auth.device.Device;
+import org.eclipse.hono.util.RegistrationConstants;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A connection event producer based on a {@link MessageSender}.
+ */
+public abstract class AbstractMessageSenderConnectionEventProducer implements ConnectionEventProducer {
+
+    private static final String MESSAGE_CONTENT_TYPE = "application/vnd.eclipse-hono-dc-notification+json";
+
+    private final HonoClient client;
+    private final BiFunction<HonoClient, String, Future<MessageSender>> messageSenderSource;
+
+    protected AbstractMessageSenderConnectionEventProducer(
```

Comment:
missing JavaDoc

the HonoClient is most likely the client for publishing the event, right? so it should probably be named something like *eventServiceClient*?